packages/circuit-ui/components/Tooltip/Tooltip.tsx

```diff
@@ -42,114 +61,85 @@ const baseStyles = ({ theme }: StyleProps) => css`
-  return value === 'left' || value === 'center' || value === 'right';
-}
-
-export type Alignment = VerticalAlignment | HorizontalAlignment;
-
-const getAlignmentStyles = ({
-  theme,
-  position,
-  align,
-}: StyleProps & TooltipProps) => {
-  const isHorizontal = position === 'bottom' || position === 'top';
+const TooltipContainer = styled.div<NoTheme>(baseStyles, typography('two'));
 
-  if (isHorizontal && isVerticalAlignment(align)) {
-    return `
-      left: 50%;
-      transform: translateX(-50%);
-
-      &::after {
-        left: 50%;
-        transform: translateX(-50%);
-      }
-    `;
-  }
-
-  if (!isHorizontal && isHorizontalAlignment(align)) {
-    return `
-      top: 50%;
-      transform: translateY(-50%);
-
-      &::after {
-        top: 50%;
-        transform: translateY(-50%);
-      }
-    `;
+const AnchoredElementWrapper = styled.button`
+  display: block;
+  border: none;
+  background-color: transparent;
+  width: 100%;
+  padding: 0px;
+  & * {
+    display: block;
   }
-
-  // align is not 'center' and therefore has the same possible values
-  // as a Position.
-  const absolutePosition = positionMap[align as Position];
-
-  /* eslint-disable max-len */
-  return `
-    ${absolutePosition}: 50%;
-    ${absolutePosition}: calc(50% - (${theme.spacings.mega} + ${theme.spacings.bit}));
-
-    &::after {
-      ${absolutePosition}: ${theme.spacings.kilo};
-    }
-  `;
-  /* eslint-enable max-len */
-};
-
-const positionAndAlignStyles = ({
-  theme,
-  position = 'right',
-  align = 'center',
-}: StyleProps & TooltipProps) => css`
-  ${getAlignmentStyles({ theme, position, align })};
-  ${getPositionStyles({ theme, position })};
 `;
 
-export interface TooltipProps {
-  position?: Position;
-  align?: Alignment;
-}
+const arrowStyles = ({ theme }: StyleProps) => css`
+  position: absolute;
+  background-color: ${theme.colors.n900};
+  width: 8px;
+  height: 8px;
+  transform: rotate(45deg);
+`;
 
-/**
- * A Tooltip component
- */
-export const Tooltip = styled.div<TooltipProps>(
-  typography('two'),
-  baseStyles,
-  shadow,
-  positionAndAlignStyles,
-);
+const Arrow = styled.div(arrowStyles);
+
+export const Tooltip = ({
+  text,
+  placement,
+  children,
+  ...props
+}: TooltipProps) => {
+  const arrowRef = useRef(null);
+  const id = uniqueId('tooltip-');
+  const {
+    x,
+    y,
+    reference,
+    placement: floatingPlacement,
+    floating,
+    strategy,
+    middlewareData: { arrow: { x: arrowX, y: arrowY } = {} },
+  } = useFloating({
+    placement,
+    whileElementsMounted: autoUpdate,
+    middleware: [offset(10), arrow({ element: arrowRef, padding: 12 })],
+  });
+  const staticSide = {
+    top: 'bottom',
+    right: 'left',
+    bottom: 'top',
+    left: 'right',
+  }[(floatingPlacement || DEFAULT_PLACEMENT).split('-')[0]] as Placement;
+
+  return (
+    <div>
+      <AnchoredElementWrapper aria-labelledby={id} ref={reference}>
+        {children}
+      </AnchoredElementWrapper>
```

Comment:
It's okay to apply the `ref` to the reference wrapper since it's only used for positioning the floating element. 

Semantic props (e.g. `aria-labelledby`) need to be applied to the reference element directly though since it's the element that can receive focus. The same goes for event listeners such as `onClick` and `onKeyDown` (how is the floating element toggled right now?).

That's why the API needs to be like the Popover's.

Reply:
To be realigned after deciding Tooltips x Toggletips